clients/da-vinci-client/src/main/java/com/linkedin/davinci/notifier/PushStatusDelayedUpdateTask.java

```diff
@@ -0,0 +1,128 @@
+package com.linkedin.davinci.notifier;
+
+import com.linkedin.venice.meta.Version;
+import com.linkedin.venice.pushmonitor.ExecutionStatus;
+import com.linkedin.venice.pushstatushelper.PushStatusStoreWriter;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * This is a delayed scheduler for sending batching push status in DaVinci.
+ */
+public class PushStatusDelayedUpdateTask {
+  private static final Logger LOGGER = LogManager.getLogger(PushStatusDelayedUpdateTask.class);
+  private Version version;
+  private boolean batchPushStartSignalSent;
+  private boolean batchPushEndSignalSent;
+  private final long daVinciWriteCompleteEventDelayInMs;
+  private final Map<Integer, ExecutionStatus> partitionStatus = new VeniceConcurrentHashMap<>();
+  // Executor for scheduling tasks
+  private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+  // Reference to the currently scheduled future task
+  private ScheduledFuture<Void> scheduledTask;
+
+  public PushStatusDelayedUpdateTask(Version version, long daVinciWriteCompleteEventDelayInMs) {
+    this.version = version;
+    this.batchPushStartSignalSent = false;
+    this.batchPushEndSignalSent = false;
+    this.daVinciWriteCompleteEventDelayInMs = daVinciWriteCompleteEventDelayInMs;
+  }
+
+  public boolean isBatchPushStartSignalSent() {
+    return batchPushStartSignalSent;
+  }
+
+  public boolean isBatchPushEndSignalSent() {
+    return batchPushEndSignalSent;
+  }
+
+  public void batchPushStartSignalSent() {
+    this.batchPushStartSignalSent = true;
+  }
+
+  public void batchPushEndSignalSent() {
+    this.batchPushEndSignalSent = true;
+  }
+
+  public void updatePartitionStatusAndMaybeSendBatchingStatus(
```

Comment:
It seems this function can be invoked by multiple threads concurrently, so shall we add any safeguard here?

Reply:
I can add synchronize keyword here; currently there is only one thread handling the status report; could you please check this in DaVinciBackend:
`private final ExecutorService ingestionReportExecutor = Executors.newSingleThreadExecutor();`

Reply:
I see.
I think adding `synchronized` is better as this thread pool size can be changed in the future.

Reply:
We probably don't need that anymore after the new change; please double check.